### DISCUSSION TITLE
Celsius and F issue Yoshan DY 2KG

### DISCUSSION THREAD

**Discussion by Archangelcpj** [artisan-roaster-scope/artisan#1550] [Q&A]
Running into an odd issue hooking to a Yoshan Roaster seems the C and F settings are crossed in the software... I have my machine connected but if I select C  in the configuration menu it displays my temps as F readings in the software and If I select F it is converting my temp settings to C...  Basically I'm running with the software in C mode and adjusting the graph to fit the F data being displayed but the graph is labeled on the side C...  If I choose F in config temp it converts my F information displayed on the roaster itself into C data and my temps show up in celcius...  Im making due by Basically ignoring the C label and us8ng it as the F data it really is...

**Comment by poundy** [comment]
i think, if not universally it's very close, PIDs ship with C as the unit of measure no matter the destination country. Artisan reads it and assumes it is whatever you say it is, and will display it as you tell it to, either with or without a conversion applied, so you need to ensure that BOTH (the temperature that Artisan reads, and the display units) are set correctly. Not having them both correct will result in either incorrect conversions, or incorrect units.  
Assuming you've done nothing to alter your PID, we expect the PID to report in C. That means, configuring the reading values to assume C. 
https://github.com/artisan-roaster-scope/artisan/blob/master/src/includes/Machines/Yoshan/DY.aset confirms input1mode=c so unless you've not used the machine configurator or changed it afterwards, you should see C in the Modbus Ports config page, like my attached image shows:
![image](https://github.com/artisan-roaster-scope/artisan/assets/279966/9f192a00-7093-4455-8ba1-67615699334d)
Then, you should use Config -> Temperature to be in the desired "display" temperature that you want.   
Do that, and retest.

**Reply by Archangelcpj** [reply]
I know my PID controllers arrived with F as the displayed temps the letter F even shows on the PID... I haven't changed that it still is set to F and indicate s so on the machine...

**Reply by poundy** [reply]
sorry, DISPLAY versus reporting are different.

**Reply by Archangelcpj** [reply]
Ok looks like with your help I've got it!!!  THANK YOU!!!  Last think I'm trying to set up my event buttons mainly for marking gas settings... Ive made the buttons label wise in the event config window to correspond with the way my Yoshan is played put basically 1KPA to 6KPA... when I mark the event a red dot shows up on my graph but it's not labeled just a red dot or little diamond...  Id like it to put a little test saying on the graph 2 kpa or something like that at the spot chronologically where I adjusted the gas... Im sure I'm missing a setting somewhere????....

**Reply by poundy** [reply]
so you've got a button that sets a value to "burner"? On the Config -> Events settings you can tick the event types you want displayed on the graph. For visibility I'd suggest you scale the values up on your button, so 2kpa would be a value of "20", 6kpa = 60.  My biggest button for gas recording is a value of 31, equivalent to my max gas of 3.1kpa. This shows as event markers in the bottom area of the graph, as well as shows "B17" at the top of the graph, indicating 1.7kpa...

**Comment by Archangelcpj** [comment]
Ahhhh OK Ill do what you suggest and get back to you THANK YOU!!!
________________________________
